file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -151,85 +151,137 @@ protected internal override async Task OnExecuteCommand()
                 else
                 {
                     logger.LogInformation("File Share Service batch create started.");
-                    batchHandle = await fileShareClient.CreateBatchAsync(buildBatchModel, CancellationToken.None);
-                    logger.LogInformation("File Share Service batch create completed for batch ID:{BatchId}.", batchHandle.BatchId);
-                    FileUploadProgress.Clear();
-                    try
+                    var createBatchResult = await fileShareClient.CreateBatchAsync(buildBatchModel, CancellationToken.None);
+                    if (createBatchResult.IsSuccess)
                     {
-                        cancellationToken.ThrowIfCancellationRequested();
-                        await Task.WhenAll(
-                        Files.SelectMany(f => f.Files.Select(file => (f, file)))
-                        .Select(f =>
+                        batchHandle = createBatchResult.Data;
+                        ExecutionResult = $"File Share Service batch create completed for batch ID: {batchHandle?.BatchId}";
+                        logger.LogInformation("File Share Service batch create completed for batch ID:{BatchId}.", batchHandle?.BatchId);
+                        bool isAddFileToBatchError = false;
+                        FileUploadProgress.Clear();
+                        try
                         {
-                            logger.LogInformation("File Share Service upload files started for file:{file} and BatchId:{BatchId} .", f.file.Name, batchHandle.BatchId);
-                            var fileUploadProgressViewModel = new FileUploadProgressViewModel(f.file.Name);
-                            FileUploadProgress.Add(fileUploadProgressViewModel);
-
-                            var (newBatchFilesViewModel, file) = f;
-                            var openRead = fileSystem.File.OpenRead(file.FullName);
-                            return fileShareClient.AddFileToBatch(batchHandle, openRead, file.Name,
-    newBatchFilesViewModel.MimeType,
-    progress =>
-    {
-                            Application.Current.Dispatcher.Invoke(() =>
-    {
-                            fileUploadProgressViewModel.CompleteBlocks = progress.blocksComplete;
-                            fileUploadProgressViewModel.TotalBlocks = progress.totalBlockCount;
-                        });
-                            if (fileUploadProgressViewModel.CompleteBlocks == fileUploadProgressViewModel.TotalBlocks)
+                            cancellationToken.ThrowIfCancellationRequested();
+                            await Task.WhenAll(
+                            Files.SelectMany(f => f.Files.Select(file => (f, file)))
+                            .Select(f =>
```

Comment:
Please convert this lambda into an async lambda to make it more readable and avoid the continuations. I have created an example PR here https://github.com/UKHO/file-share-service-desktop-client/pull/35